cli/internal/helm/client.go

```diff
@@ -269,26 +278,38 @@ func (c *Client) upgradeRelease(
 		releaseName = constellationServicesInfo.releaseName
 		values, err = loader.loadConstellationServicesValues()
 
-		// TODO: v2.8: remove this call.
-		// Manually setting attestationVariant is required here since upgrade normally isn't allowed to change this value.
-		// However, to introduce the value into a 2.6 cluster for the first time we have to set it nevertheless.
-		if err := setAttestationVariant(values, conf.AttestationVariant); err != nil {
-			return fmt.Errorf("setting attestationVariant: %w", err)
-		}
-
 		// TODO: v2.8: remove from here...
```

Comment:
Since there are multiple different migration taking place here, I'd move each migration to its on function and guard it at the beginning of the function by actually checking if we have an upgrade where we need to migrate, so that even if we forget deleting the code, the migration is never applies twice. 

For example for the new migration added in this PR we can actually check if we upgrade from `v2.6.A` to `v2.7.B`. Then upgrading to `v2.7.C` or `v2.8.D` will not trigger the migration again.

Reply:
Changed code so that migrations can be left in the codebase:
- Moved migrations into a separate `applyMigrations` function. 
- Added `migrateFrom2_6` function that is called based on the "from" version of the upgrade. This way the migration is only called where required.